VirtualFinland.Infrastructure/Features/SimpleEmailService.cs

```diff
@@ -0,0 +1,257 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Pulumi;
+using Pulumi.Aws.Route53;
+using Pulumi.Aws.Ses;
+using Pulumi.Aws.Ses.Inputs;
+using VirtualFinland.Infrastructure.Common;
+
+namespace VirtualFinland.Infrastructure.Features;
+
+public class SimpleEmailService
+{
+    private readonly StackSetup _setup;
+    private readonly string? _domainZoneId;
+    private readonly string? _domainName;
+    private readonly string _mailFromSubDomain;
+    private readonly bool _createDomainRecords;
+    private readonly List<StackOwnsDomain> _createDomainRecordsFor = new();
+
+    public DomainIdentity? DomainIdentity { get; private set; }
+    public DomainDkim? DomainDkim { get; private set; }
+
+    public SimpleEmailService(StackSetup setup)
+    {
+        _setup = setup;
+
+        var config = new Config("ses");
+        _domainName = config.Get("domain-name") ?? "";
+        _mailFromSubDomain = config.Get("mail-from-sub-domain") ?? "ses";
+        _domainZoneId = config.Get("domain-zone-id") ?? null;
+        _createDomainRecords = config.GetBoolean("create-domain-records") ?? false;
+
+        if (_createDomainRecords)
+        {
+            _createDomainRecordsFor.Add(new StackOwnsDomain
+            {
+                StackName = setup.Environment,
+                DomainName = _domainName,
+            });
+
+            var createDomainRecordsFor = config.GetObject<List<string>>("also-create-domain-records-for");
+            if (createDomainRecordsFor != null)
+            {
+                foreach (var stackOwnsDomain in createDomainRecordsFor)
+                {
+                    var stackOwnsDomainParts = stackOwnsDomain?.Split(':');
+                    if (stackOwnsDomainParts != null && stackOwnsDomainParts.Length == 2)
+                    {
+                        if (stackOwnsDomainParts[0] == setup.Environment)
+                            throw new ArgumentException($"The current stack {setup.Environment} can't be defined in the ses:also-create-domain-records-for list");
```

Comment:
Heh, vielä kiusaan sen verran että tätä ei pitäisi varmaan tapahtua, ellei pulumin conffiin listaisi vahingossa "itseään", eli vaikka mvp-staging listaisi itsensä myös `ses:also-create-domain-records-for` ?

Hyvähän se on tietysti olla varuillaan, tyhmät virheet ovat todellisia.

Reply:
Jes, sisäisestä siisteydestä voi olla kiitollinen kun tulee stoppi kun kopipastettelee konffeja ja kovaa ajoa :)